examples/spack/Dockerfile

```diff
@@ -59,23 +62,30 @@ RUN spack compiler find --scope system
 # Test: Some basic commands.
 RUN which spack
 RUN spack --version
+RUN spack compiler find
 RUN spack compiler list
 RUN spack compiler list --scope=system
 RUN spack compiler list --scope=user
 RUN spack compilers
-RUN spack spec netcdf
+RUN spack spec charliecloud
+
+# Add system gpg to packages.yaml to avoid the gpg filename too long error. See
+# https://github.com/spack/spack/issues/14605
+RUN ( echo "  gnupg:" \
+ &&   echo "    paths:" \
+ &&   echo "      gnupg@2.0.22: /usr" \
+ &&   echo "    buildable: false" ) \
+       >>/spack/etc/spack/defaults/packages.yaml
 
 # Test: Install Charliecloud.
-RUN spack spec charliecloud+builder
-RUN spack install charliecloud+builder
+RUN spack spec charliecloud+builder ^python+libxml2
+RUN spack install charliecloud+builder ^python+libxml2
```

Comment:
This should be addressed in the charliecloud package.

@nicholas-sly can you comment on why we need to add the `^python+libxml2` dependency?

Reply:
This is a factor of a circular dependency resolution between libxml2 and python, I believe.  The hope is that the new concretizer (still a couple months away) will fix this issue and remove that variant, but for now, it is necessary to build.